plugin/src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -350,6 +350,27 @@ public void deleteIsland(@NotNull Island island) {
         });
     }
 
+    /**
+     * Gets an Island upgrade
+     *
+     * @param island  The specified Island
+     * @param upgrade The specified Upgrade's name
+     * @return The island Upgrade
+     */
+    public IslandUpgrade getIslandUpgrade(@NotNull Island island, @NotNull String upgrade) {
+        Optional<IslandUpgrade> islandUpgrade =
+                IridiumSkyblock.getInstance().getDatabaseManager().getIslandUpgradeTableManager().getEntries().stream().filter(
+                        isUpgrade -> island.equals(isUpgrade.getIsland().orElse(null)) && isUpgrade.getUpgrade().equalsIgnoreCase(upgrade)
+                ).findFirst();
+        if (islandUpgrade.isPresent()) {
+            return islandUpgrade.get();
+        } else {
+            IslandUpgrade isUpgrade = new IslandUpgrade(island, upgrade);
```

Comment:
Unnecessary abbreviation

Reply:
You mean for the name? No because we already have something called islandUpgrade?

Reply:
If you have two variables that could share the same name, find a difference between them. It doesn't help that one variable uses an abbreviation while the other one doesn't.

Reply:
im bad with names so ima just say :shrug: